.github/workflows/check_api_docs.yml

```diff
@@ -0,0 +1,42 @@
+name: check-api-docs
+on:
+  workflow_dispatch:
+    inputs:
+      project:
+        description: 'Project to check'
+        required: true
+        default: 'csip'
+        type: choice
+        options:
+          - alerts
+          - csip
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Check API docs
+        id: check-api
+        run: ./check-api-docs.bash ${{ inputs.project }}
+
+      - name: Upload differences report
+        if: check-api.outputs.differences == 'true'
+        uses: actions/upload-artifact@v4
+        with:
+          path: build/api.diff
+
+      - name: Create Pull Request
+        if: check-api.outputs.differences == 'true'
+        uses: peter-evans/create-pull-request@v7
```

Comment:
Bit surpsied that a 3rd party action is required here - I would have thought Github would have their own action that just wraps the gh CLI or something.